### ISSUE TITLE
titanic prediction

### ISSUE THREAD

**Issue by raj1209-2910** [raj1209-2910/titanic#1]
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import math
% matplotlib in line
titanic = pd.read_csv('train.csv')
titanic.head()
titanic.shape
sns.countplot(x='Survived', data = titanic)
sns.countplot(x='Survived', hue= 'Sex', data= titanic, palette='winter')
sns.countplot(x='Survived', hue='Pclass', data=titanic, palette='PuBu')
titanic['Age'].plot.hist()
titanic['Fare'].plot.hist(bins = 20, figsize =(10,5))
titanic['Parch'].plot.hist()
sns.countplot(x='Parch', data=titanic, palette='summer')
titanic.isnull().sum()
sns.heatmap(titanic.isnull(), cmap='spring')
sns.boxplot(x='Pclass',y='Age', data=titanic)
titanic.head()
titanic.drop('Cabin', axis=1, inplace=True)
titanic.head(3)
titanic.dropna(inplace = True)
sns.heatmap(titanic.isnull(), cbar=False)
titanic.isnull().sum()
pd.get_dummies(titanic['Sex']).head()
sex= pd.get_dummies(titanic['Sex'],drop_first=True)
sex.head()
embark= pd.get_dummies(titanic['Embarked'])
embark.head(3)
embark= pd.get_dummies(titanic['Embarked'], drop_first=True)
embark.head(3)
Pcl= pd.get_dummies(titanic['Pclass'], drop_first=True)
Pcl.head(3)
titanic = pd.concat([titanic, sex, embark, Pcl], axis = 1)
titanic.head(3)
X=titanic.drop('Survived', axis=1)
y= titanic['Survived']
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,y, test_size=0.2, random_state = 2)
print(X.shape,X_train.shape,X_test.shape)
from sklearn.linear_model import LogisticRegression
model = LogisticRegression()
model.fit(X_train, y_train)
prediction = model.predict(X_test)
from sklearn.metrics import classification_report
from sklearn.metrics import confusion_matrix
confusion_matrix(y_test, prediction)
from sklearn.metrics import accuracy_score
accuracy_score(y_test, prediction)
